Test error in 0.16.x after merging

## Summary

After merging #4942 there's a test failure... don't know if it's random or permanent

## Branch or installer method


 - Branch: 0.16.x


## Traceback or relevant snippet from server.log

https://circleci.com/gh/learningequality/ka-lite/2045?utm_campaign=build-failed&utm_medium=email&utm_source=notification

```

FAIL: test_playlist_progress (kalite.coachreports.tests.coachreport_tests.PlaylistProgressResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/ka-lite/kalite/coachreports/tests/coachreport_tests.py", line 141, in test_playlist_progress
    self.assertEqual(len(exercises),1)
AssertionError: 2 != 1
```

There is no conceivable way that these two are related, but we should sort
it.

On Thu, 3 Mar 2016, 03:58 Benjamin Bach, <notifications@github.com> wrote:

> Summary
>
> After merging #4942
> <https://github.com/learningequality/ka-lite/pull/4942> there's a test
> failure... don't know if it's random or permanent
> Branch or installer method
>
> *Please specify if you are using the current develop branch or which
> installation source you were using when the issue occurred.*
>
>    - Branch: 0.16.x
>
> Traceback or relevant snippet from server.log
>
>
> https://circleci.com/gh/learningequality/ka-lite/2045?utm_campaign=build-failed&utm_medium=email&utm_source=notification
>
>
> FAIL: test_playlist_progress (kalite.coachreports.tests.coachreport_tests.PlaylistProgressResourceTestCase)
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "/home/ubuntu/ka-lite/kalite/coachreports/tests/coachreport_tests.py", line 141, in test_playlist_progress
>     self.assertEqual(len(exercises),1)
> AssertionError: 2 != 1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/learningequality/ka-lite/issues/4956>.
>

Persistent, think it is related to a new content_pack.

```
======================================================================
FAIL: test_playlist_progress (kalite.coachreports.tests.coachreport_tests.PlaylistProgressResourceTestCase)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/ka-lite/kalite/coachreports/tests/coachreport_tests.py", line 141, in test_playlist_progress
    self.assertEqual(len(exercises),1)
AssertionError: 2 != 1

======================================================================
FAIL: test_topic_availability (kalite.updates.tests.availability_tests.TestTopicAvailability)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/ka-lite/kalite/updates/tests/availability_tests.py", line 23, in test_topic_availability
    self.assertEqual(topic["available"], any_available, "Topic availability for {topic} did not match child availability when any children are available.".format(topic=topic.get("title")))
AssertionError: False != True : Topic availability for Comparing two-digit numbers did not match child availability when any children are available.

======================================================================
FAIL: test_explore_overall (kalite.topic_tools.tests.explore_tests.TestExploreMethods)
get_explore_recommendations()
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ubuntu/ka-lite/kalite/topic_tools/tests/explore_tests.py", line 63, in test_explore_overall
    self.assertEqual(actual[0].get("interest_topic").get("id"), "in-sixth-grade-math")
AssertionError: u'arithmetic' != 'in-sixth-grade-math'

----------------------------------------------------------------------
```

--- a/kalite/coachreports/tests/coachreport_tests.py
+++ b/kalite/coachreports/tests/coachreport_tests.py
@@ -115,32 +115,43 @@ class PlaylistProgressResourceTestCase(FacilityMixins, StudentProgressMixin, KAL
     @set_database
     def setUp(self, db=None):
         self.db = db
-        self.student=self.create_student()
-        self.ex_logs=self.create_exercise_log(user=self.student)
-        self.Parent=get_content_parents(ids=[self.ex_logs.exercise_id])
-        self.p=self.Parent[0]
-        q=Item.update(available=True).where(Item.id == self.p['id'])
-        q.execute()
-        q1=Item.update(available=True).where(Item.id == self.ex_logs.exercise_id)
-        q1.execute()
+        self.topic1 = create({
+            "id": "stuff_here",
+            "slug": "stuff_here",
+            "path": "qwrqweqweqwe",
+            "kind": "Topic",
+            "title": "",
+            "description": "",
+            "available": True,
+        })
+        self.exercise1 = create({
+            "id": "stuff_here_ex",
+            "slug": "stuff_here_ex",
+            "path": "qwrqweqweqwe/qwrqweqweq",
+            "kind": "Exercise",
+            "title": "",
+            "description": "",
+            "available": True,
+            "parent": self.topic1,
+        })
+        self.student = self.create_student()
+        self.ex_logs = self.create_exercise_log(user=self.student, exercise_id=self.exercise1.id)
 
     @set_database
     def tearDown(self, db=None):
         self.db = db
-        q=Item.update(available=False).where(Item.id == self.p['id'])
-        q.execute()
-        q1=Item.update(available=False).where(Item.id == self.ex_logs.exercise_id)
-        q1.execute()
+        self.exercise1.delete()
+        self.topic1.delete()
 
     def test_playlist_progress(self):
         base_url = self.reverse('api_dispatch_list', kwargs={'resource_name': 'playlist_progress_detail'})
-        url = base_url + '?' + urllib.urlencode({'user_id':self.student.id, 'playlist_id': self.p["id"]})
+        url = base_url + '?' + urllib.urlencode({'user_id': self.student.id, 'playlist_id': self.topic1.id})
         resp = self.client.get(url)
-        exercises=json.loads(resp.content).get("objects")
+        exercises = json.loads(resp.content).get("objects")
         # checking if the request returned any of the exercises
-        self.assertEqual(len(exercises),1)
+        assert len(exercises)
         #checking if the returned list is accurate
-        self.assertEqual(exercises[0]["title"],"Comparing two-digit numbers")
+        self.assertEqual(exercises[0]["title"], self.exercise1.title)
 
 
 class LearnerLogAPITests(FacilityMixins,

--- a/kalite/coachreports/tests/coachreport_tests.py
+++ b/kalite/coachreports/tests/coachreport_tests.py
@@ -149,7 +149,7 @@ class PlaylistProgressResourceTestCase(FacilityMixins, StudentProgressMixin, KAL
         resp = self.client.get(url)
         exercises = json.loads(resp.content).get("objects")
         # checking if the request returned any of the exercises
-        assert len(exercises)
+        self.assertTrue(len(exercises) > 0)
         #checking if the returned list is accurate
         self.assertEqual(exercises[0]["title"], self.exercise1.title)
 

--- a/kalite/distributed/static/js/distributed/topics/views.js
+++ b/kalite/distributed/static/js/distributed/topics/views.js
@@ -193,7 +193,7 @@ var SidebarView = BaseView.extend({
         this.render();
 
         this.listenTo(this.state_model, "change:open", this.update_sidebar_visibility);
-        this.listenTo(this.state_model, "change:current_level", this.resize_sidebar);
+        this.listenTo(this.state_model, "change:current_level", this.update_sidebar_visibility);
     },
 
     render: function() {
@@ -236,7 +236,7 @@ var SidebarView = BaseView.extend({
         }
     },
 
-    resize_for_narrow: _.throttle(function() {
+    resize_for_narrow: function() {
         if (this.state_model.get("open")) {
             var current_level = this.state_model.get("current_level");
             var column_width = this.$(".topic-container-inner").width();
@@ -276,23 +276,23 @@ var SidebarView = BaseView.extend({
 
             this.sidebarTab.animate({left: this.sidebar.width() + sidebarPanelNewLeft}, 115);
         }
-    }, 100),
+    },
 
     // Pretty much the code for pre-back-button sidebar resize
-    resize_for_wide: _.throttle(function() {
-        var current_level = this.state_model.get("current_level");
-        var column_width = this.$(".topic-container-inner").width();
-        var last_column_width = 400;
-        
-        this.width = (current_level-1) * column_width + last_column_width + 10;
-        this.sidebar.width(this.width);
-        if (this.state_model.get("open")) {
-            this.sidebar.css({left: 0});
-            this.sidebarTab.css({left: this.width});
-        }
+    resize_for_wide: function() {
+       if (this.state_model.get("open")) {
+           var current_level = this.state_model.get("current_level");
+           var column_width = this.$(".topic-container-inner").width();
+           var last_column_width = 400;
 
-        this.set_sidebar_back();
-    }, 100),
+           this.width = (current_level-1) * column_width + last_column_width + 10;
+           this.sidebar.width(this.width);
+           this.sidebar.css({left: 0});
+           this.sidebarTab.css({left: this.width});
+
+            this.set_sidebar_back();
+       }
+    },
 
     check_external_click: function(ev) {
         if (this.state_model.get("open")) {
@@ -312,10 +312,10 @@ var SidebarView = BaseView.extend({
 
     update_sidebar_visibility: _.debounce(function() {
         if (this.state_model.get("open")) {
-            // Used to get left value in number form
-            var sidebarPanelPosition = this.sidebar.position();
             this.sidebar.css({left: 0});
             this.resize_sidebar();
+            // Used to get left value in number form
+            var sidebarPanelPosition = this.sidebar.position();
             this.sidebarTab.css({left: this.sidebar.width() + sidebarPanelPosition.left}).html('<span class="icon-circle-left"></span>');
             this.$(".sidebar-fade").show();
         } else {
@@ -372,11 +372,15 @@ var SidebarView = BaseView.extend({
     },
 
     show_sidebar: function() {
-        this.state_model.set("open", true);
+        if (!this.state_model.get("open")) {
+            this.state_model.set("open", true);
+        }
     },
 
     hide_sidebar: function() {
-        this.state_model.set("open", false);
+        if (this.state_model.get("open")) {
+            this.state_model.set("open", false);
+        }
     },
 
     show_sidebar_tab: function() {

--- a/kalite/topic_tools/tests/explore_tests.py
+++ b/kalite/topic_tools/tests/explore_tests.py
@@ -60,4 +60,4 @@ class TestExploreMethods(KALiteTestCase):
 
         actual = get_explore_recommendations(self.user1, request)
 
-        self.assertEqual(actual[0].get("interest_topic").get("id"), "in-sixth-grade-math")
+        self.assertEqual(actual[0].get("interest_topic").get("id"), "arithmetic")

